roles/odf_setup/templates/openshift-storage-cluster.yml.j2

```diff
@@ -16,7 +16,7 @@ spec:
   manageNodes: false
   monDataDirHostPath: /var/lib/rook
   storageDeviceSets:
-  - count: {{ local_storage_devices | length }}
+  - count: {{ local_storage_devices | default([1]) | length }}
```

Comment:
let's set the default of `local_storage_devices` in the default/main.yml directory instead. 

Because of the naming convention we most likely need to update this variable to use the role prefix, that would be either `odf_setup_`  or `os_`  any of them are fine.

Reply:
Well, if we set the default value in `default/main.yml` then it will also affect this part: https://github.com/semihkisa/ansible-collection-redhatci-ocp/blob/update_odf_role/roles/odf_setup/templates/ocs-disk-gatherer.yml.j2#L84

Which I am not sure if this is the intention. The reason why I assigned a default value there is just to have minimum count for `storageDeviceSets`. So I'd rather set this variable a required one and don't set a default device list. What do you think?

Reply:
> Well, if we set the default value in `default/main.yml` then it will also affect this part: https://github.com/semihkisa/ansible-collection-redhatci-ocp/blob/update_odf_role/roles/odf_setup/templates/ocs-disk-gatherer.yml.j2#L84
> 
> Which I am not sure if this is the intention. The reason why I assigned a default value there is just to have minimum count for `storageDeviceSets`. So I'd rather set this variable a required one and don't set a default device list. What do you think?

if there is a required minimum that it absolutely needs to work, then maybe the variable should be reworked into something like `extra_storage_devices` which defaults to 0/none/empty. Then in the template you would add the required 1 + however many extra devices.

Reply:
@semihkisa I think we should make the `local_storage_devices` required and assert that it is a list of at least 1 item. 

Sorry for the delay, I was checking how to validate this on our lab and the PTO. Cheers.

Reply:
Np at all @betoredhat, I was also thinking that this var should be required so now I made the change to make it required and added a small step in the validation.